torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2239,14 +2240,15 @@ c10::intrusive_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
       device,
       rank,
       opType,
-      seq_,
+      seqCollective_,
```

Comment:
Will WorkNCCL only be collective work, not P2P work?

Reply:
> does it make sense to pass seqCollective_ to seqP2P_ NCCL work?

That's a good point @shuqiangzhang!

@wconstab / @zdevito - do we need to differentiate p2p v/s collectives at the NCCL layer? I think the answer is yes - but wanted your thoughts.

I'll address this in a follow on pull request

Reply:
WorkNCCL is not part of the 'NCCL layer' (as in, inside nccl library), it's part of pytorch pg-nccl layer still.  It's just a container we use to represent that there is a work object that's being tracked by our watchdog.  It should match up with our flight recorder.  When the watchdog reports a timeout of some operation at seq_id blah, that should correspond with FR dumps.